### PR TITLE
Use recommended addon-resizer 1.8.4

### PR DESCRIPTION
**Pull request by goll**
PR https://github.com/coreos/prometheus-operator/pull/2380 updated addon-resizer but not to the current recommended version: 

https://github.com/kubernetes/autoscaler/tree/master/addon-resizer#addon-resizer
`Currently recommended version is 1.8, on addon-resizer-release-1.8 branch.`

`addon-resizer-amd64:2.1` has a timestamp of `Aug 10, 2017`
`addon-resizer:1.8.4` has a timestamp of `Nov 9, 2018`

Searching through [kubernetes](https://github.com/kubernetes/kubernetes/search?q=addon-resizer&type=Code) there are multiple references to `k8s.gcr.io/addon-resizer:1.8.4` in the current codebase.

Example for `kube-state-metrics`: [kube-state-metrics-deployment.yaml](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/prometheus/kube-state-metrics-deployment.yaml#L41)

Bump docs and `alertmanager` to `v0.16.1`.

Rename `Watchdog` alert to `DeadMansSwitch` as the docs reference the latter.

Apologies for the diff size, followed the instructions with:
```
$ jb update
$ make clean generate-in-docker
```

### CODE REVIEW

**Comment by goll** [comment]
Also [travis](https://github.com/coreos/prometheus-operator/blob/master/scripts/create-minikube.sh#L11-L12) is using deprecated versions of minikube/kubernetes:
```
export MINIKUBE_VERSION=v0.28.0
export KUBERNETES_VERSION=v1.10.0
```
The changes were tested locally with `minikube=v0.35.0` using `kubernetes=v1.13.4`.

**Comment by brancz** [comment]
> Rename Watchdog alert to DeadMansSwitch as the docs reference the latter.

We intentionally renamed this, but seem to have forgotten to change the docs. Could you take care of that? :slightly_smiling_face: 

@mxinden I believe you recently looked into the addon-resizer things, can you give this a quick review as well? Thanks!

File: contrib/kube-prometheus/jsonnet/kube-prometheus/alerts/general.libsonnet [hunk 1]
```diff
@@ -23,7 +23,7 @@
                 This alert is always firing, therefore it should always be firing in Alertmanager
                 and always fire against a receiver. There are integrations with various notification
                 mechanisms that send a notification when this alert is not firing. For example the
-                "DeadMansSnitch" integration in PagerDuty.
```
Comment (brancz) [inline contrib/kube-prometheus/jsonnet/kube-prometheus/alerts/general.libsonnet]: This PagerDuty integration is actually called `DeadMansSnitch`, so this was correct.

Comment (goll) [reply contrib/kube-prometheus/jsonnet/kube-prometheus/alerts/general.libsonnet]: :man_facepalming: sorry about that, I assumed that was a typo, fixing

**Comment by brancz** [comment]
lgtm :+1: thanks!

**Pull request merged**